Tests/NIOTransportServicesTests/NIOTSBootstrapTests.swift

```diff
@@ -371,6 +371,36 @@ final class NIOTSBootstrapTests: XCTestCase {
         XCTAssertEqual(try listenerChannel.getOption(NIOTSChannelOptions.multipathServiceType).wait(), .handover)
         XCTAssertEqual(try connectionChannel.getOption(NIOTSChannelOptions.multipathServiceType).wait(), .handover)
     }
+
+    func testNWParametersConfigurator() throws {
+        let group = NIOTSEventLoopGroup()
+        defer {
+            try! group.syncShutdownGracefully()
+        }
+
+        let configuratorListenerCounter = NIOLockedValueBox(0)
+        let configuratorConnectionCounter = NIOLockedValueBox(0)
+
+        let listenerChannel = try NIOTSListenerBootstrap(group: group)
+            .configureNWParameters { _ in
+                configuratorListenerCounter.withLockedValue { $0 += 1 }
+            }
+            .bind(host: "localhost", port: 0)
+            .wait()
+
+        let connectionChannel: Channel = try NIOTSConnectionBootstrap(group: group)
+            .configureNWParameters { _ in
+                configuratorConnectionCounter.withLockedValue { $0 += 1 }
+            }
+            .connect(to: listenerChannel.localAddress!)
+            .wait()
+
+        try listenerChannel.close().wait()
+        try connectionChannel.close().wait()
+
+        XCTAssertEqual(1, configuratorListenerCounter.withLockedValue { $0 })
```

Comment:
Given the configurator is currently shared between the listening channel and accepted server channels, shouldn't the count here be two? (one for the listener, one for the accepted channel)

Reply:
There are separate counters for each configurator in the tests, so they will be 1 each.

The listener channel will create connection channels and pass down the configurator, but that happens when the `newConnectionHandler` is called by the underlying `NWListener`, which I'm not doing in this test.

Reply:
`newConnectionHandler` should be getting called:
- You start a server which listens for connections (+1 listener counter)
- You start a client which opens a connection to the server (+1 connection counter)
- The server accepts that new connection and `newConnectionHandler` should be called so we'd expect a +1 listener counter.